_meeting-notes/2019-12-10.md

```diff
@@ -0,0 +1,150 @@
+
+- @raul feedback on the Critical Path Tree Feedback
+  - Break Infra into
+    - Infra structure
+      - 
+    - Infra as a Service
+      - 1st and 2nd sticky
+  - AI(David) Augment DHT Plan with test cases for the "critical patches made to the DHT"
+    - https://github.com/libp2p/go-libp2p-kad-dht/issues/345
+
+- @anton We've hit Docker Swarm scalability issues
+  - @raul - try simpler docker image - just `ping`
+  - @jim - try parallelism config of docker swarm (start with fewer replicas)
+
+## :dart: Next
+
+~~### OKRs for the Quarter~~ #1 Goal: Unblock go-ipfs 0.5.0
+~~https://raw.githubusercontent.com/ipfs/testground/master/docs/img/Q4OKRs.png~~
+~~### Top 3 things to get done this week~~
+
+We are putting aside the OKRs for the time being and instead **focus on the #1 Goal: Unblock go-ipfs 0.5.0**. This goal has a very clear list of things that need to ship:
+
+- [ ] Instrument go-libp2p so that the DHT operations are observable and traced in a Distributed Context (owner: @raulk)
+- [ ] Add support to Traffic Shaping & Network Configuration to TestGround (owner: @Stebalien)
+- [ ] Expand the TestGround Infrastructure so that it can run multiple thousands of nodes (owner: @nonsense)
+- [ ] Creation of a TestGround Dashboard that can aid the developers of go-ipfs & go-libp2p to grok the results outputted by every node in a TestGround run (owner: @daviddias)
+- [ ] Keep on iterating the DHT Test Plan (@owner: @daviddias)
+- [ ] Design blueprint for multi-version tests (aka test ensembles) (owner: @raulk)
+
+Full list can be at https://github.com/ipfs/testground/issues/196
+
+
+
+### Assignments
+
+- ==@raulk==
+    - (HIGH PRIO, working on this now) Outline candidate solutions and trade-offs for libp2p deep traceability.
+    - Proof of concept of 3 different solutions that Raúl is thinking of (OpenTracing with mock tracer, diagnostics context, go-log tracer).
+- ==@daviddias==
+  - Host Testground Hack week
+  - Study the current situation of go-libp2p metrics and report back
+    - See how open-census is working there
+    - Experiment with jaeger 
+  - Review @jimpick PR https://github.com/ipfs/testground/pull/232
+- ==@nonsense==
+  - Measure the memory footprint of a go-libp2p node that runs 1000 find peer queries on the mainnet
+  - Use that measure to calculate how many nodes can be run in each AWS instance
+  - Calculate the right number of machines needed for a cluster to run 10000 libp2p nodes
+  - Enhance the Testground experience by removing the overhead of running a plan (devs should not have to worry about infra). Running a plan should be done in one command
+      - @nonsense - this assumes we run a service on AWS and solve issues like:
+          - https://github.com/ipfs/testground/issues/226
+  - Get filebeat to only pump logs at the end of a run (so that it doesn't influence the performance of the test)
+      - @nonsense - this seems to be the lowest priority for the week, right?
+- ==@hacdias==
```

Comment:
@hacdias can you also take a look at this list at https://github.com/ipfs/testground/pull/233/files#r356468557 and pick some of those to get done this week?

Reply:
From https://github.com/ipfs/testground/pull/233/files#r356514633, @hacdias is now owning
- #191
- #124 
- #158